tests/unit/teardown-context-test.js

```diff
@@ -65,28 +64,16 @@ module('teardownContext', function (hooks) {
     assert.strictEqual(getContext(), undefined, 'context is unset');
   });
 
-  test('destroyables registered with the context are invoked', async function (assert) {
-    registerDestructor(context, () => {
```

Comment:
Yeah I think this assertion is the tricky one. If the desire is that destructors can be registered on the test context and are cleaned up, then we might need an option `teardownContext(context, { destroyContext: true });` which is called during from the `afterEach` implementation.

But I think decoupling any of the test suites _own_ setup/teardown from destructors is still the right move.

Basically if this test needs to pass, or if destructors need to be registered on the owner and torn down at the end of an Ember test and that is already considered public API, IMO it can be made to work.